classes/core/APIRouter.inc.php

```diff
@@ -77,7 +80,16 @@ function route($request) {
 		// Ensure slim library is available
 		require_once('lib/pkp/lib/vendor/autoload.php');
 
-		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $this->getEntity());
+		$sourceFile = null;
+		$entity = $this->getEntity();
+		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $entity[0]);
+
+		if (!empty($entity[1])) {
```

Comment:
Alright. Let's say we have an incoming request with the following uri: `api/v1/alpha/beta/gamma/1/omega`, How would you approach this? Could you please suggest an entity identification strategy?